sid/script.js

```diff
@@ -0,0 +1,633 @@
+    }
+}
+
+function reset() {
+    setTimeout(function () {
+        notificationOne.textContent = '';
+        notificationTwo.textContent = '';
+        notificationThree.textContent = '';
+        mainHeading.style.opacity = '0';
+        mainHeading.style.backgroundColor = '';
+        mainHeading.style.transitionDuration = '1s';
+        mainHeading.style.transitionTimingFunction = 'ease-in';
+        document.body.style.backgroundImage = 'none';
+        document.body.style.transitionDuration = '1s';
+        document.body.style.transitionTimingFunction = 'ease-in';
+        for (const button of allButtons) {
+            button.style.backgroundColor = '';
+            button.style.color = 'white';
+        }
+        for (const block of allBlocks) {
+            block.style.backgroundColor = '';
+            block.textContent = '';
+            block.style.border = 'grey solid 1px';
+            block.style.color = 'white';
+        }
+    }, 1000);
+    setTimeout(function () {
+        mainHeading.textContent = 'Wordle Time!';
+        mainHeading.style.color = 'white';
+        mainHeading.style.opacity = '1';
+        mainHeading.style.backgroundColor = 'none';
+        mainHeading.style.transitionDuration = '1s';
+        mainHeading.style.transitionTimingFunction = 'ease-in';
+        scoreSection.style.fontSize = '1.25em';
+    }, 2000);
+    document.body.style.backgroundImage = '';
+    moreChances.style.display = 'none';
+    checkFullSquares();
+    imageFadeOut(img);
+    imageFadeOut(img2);
+    blockNumber = 0;
+    gameComplete = false;
+    pressedEnter = false;
+    wordChecked = false;
+    found = false;
+    rowChanged = false;
+    nextStartingPoint = 0;
+    rowNumber = -1;
+    rowsAdded = false;
+    gameLost = false;
+    random = dictionary[Math.floor(Math.random() * dictionary.length)];
+    console.log(random);
+    const allDivs = document.querySelectorAll('div');
+    for (const row of allDivs) {
+        if (row.classList.contains('newRow')) {
+            row.remove();
+        }
+    }
+    allBlocks = document.querySelectorAll('.squares');
+    rows = document.querySelectorAll('.row');
+}
+
+function checkFullSquares() {
+    for (const block of allBlocks) {
+        if (block.textContent !== '') {
+            block.style.border = '#fcfcfc solid 4px';
+        }
+        else {
+            block.style.border = 'grey solid 1px';
+        }
+    }
+}
+
+function wordInDict(word) {
+    if (dictionary.includes(word)) {
+        return true;
+    }
+    else {
+        return false;
+    }
+}
+
+function wordFound(event) {
+    if (gameComplete === false) {
+        if (event.type === 'click') {
+            wordTypeOut(event.target.textContent);
+        }
+        if (event.type === 'keyup' && event.key.length === 1 && event.key.match(/[a-z]/i) || event.key === 'Enter' || event.key === 'Backspace') {
+            buttonPress(event.key);
+            wordTypeOut(event.key);
+        }
+    }
+}
```

Comment:
I think, it'd be clearer if we split `wordFound` function into two functions and attach them to appropriate events, e.g.

```suggestion
function handleVirtualKeyPress(event) {
    if (!gameComplete) {
        wordTypeOut(event.target.textContent);        
    }
}

function handlePhysicalKeyPress(event) {
    if (gameComplete) {
        return;
    }
    
    if (event.key.match(/^([a-z]|enter|backspace)$/i)) {
        buttonPress(event.key);
        wordTypeOut(event.key);
    }
}
```